client/state/sharing/services/selectors.js

```diff
@@ -9,6 +19,58 @@ export function getKeyringServices( state ) {
 }
 
 /**
+ * Returns an object of service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getKeyringServicesByType( state, type ) {
+	return filter( getKeyringServices( state ), { type: type } );
+}
+
+/**
+ * Returns an object of eligible service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {Object} site  Site object.
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getEligibleKeyringServices( state, site, type ) {
+	const services = getKeyringServicesByType( state, type );
+
+	if ( ! site ) {
+		return services;
+	}
+
+	return services.filter( ( service ) => {
+		// Omit if the site is Jetpack and service doesn't support Jetpack
+		if ( site.jetpack && ! service.jetpack_support ) {
+			return false;
+		}
+
+		// Omit if Jetpack module not activated
+		if ( site.jetpack && service.jetpack_module_required &&
+			! site.isModuleActive( service.jetpack_module_required ) ) {
```

Comment:
You should use the [`isJetpackModuleActive`](https://github.com/Automattic/wp-calypso/blob/08b32a64728927babb7b093ba988e695c37eaafd/client/state/sites/selectors.js#L136-L153) selector here.

Note also that the site object you're passing in has been decorated with methods such as `isModuleActive`, which do not exist on a site from Redux state. This is legacy and we should not rely on it to exist much longer.